spec/presenters/vendor_api/single_application_presenter_spec.rb

```diff
@@ -260,7 +260,58 @@
 
       response = VendorAPI::SingleApplicationPresenter.new(application_choice).as_json
 
-      expect(response.dig(:attributes, :candidate, :uk_residency_status)).to eq('Candidate does not know')
+      expect(response.dig(:attributes, :candidate, :uk_residency_status)).to eq('Candidate needs to apply for permission to work and study in the UK')
+    end
+  end
+
+  describe 'uk_residency_status_code' do
+    it 'returns A if one of the candidate nationalities is GB' do
+      application_form = build_stubbed(:application_form,
+                                       :minimum_info,
+                                       first_nationality: 'Irish',
+                                       second_nationality: 'British')
+      application_choice = build_stubbed(:application_choice, status: 'awaiting_provider_decision', application_form: application_form)
+
+      response = VendorAPI::SingleApplicationPresenter.new(application_choice).as_json
+
+      expect(response.dig(:attributes, :candidate, :uk_residency_status_code)).to eq('A')
+    end
+
+    it 'returns B if one of the candidate nationalities is IE' do
+      application_form = build_stubbed(:application_form,
+                                       :minimum_info,
+                                       first_nationality: 'Canadian',
+                                       second_nationality: 'Irish')
+      application_choice = build_stubbed(:application_choice, status: 'awaiting_provider_decision', application_form: application_form)
+
+      response = VendorAPI::SingleApplicationPresenter.new(application_choice).as_json
+
+      expect(response.dig(:attributes, :candidate, :uk_residency_status_code)).to eq('B')
+    end
+
+    it 'returns C if the candidate does not have residency or right to work in UK' do
```

Comment:
could you add a test for when it could be `decide_later` just to ensure its we've covered the other option?